src/java.base/share/classes/java/lang/invoke/InnerClassLambdaMetafactory.java

```diff
@@ -623,7 +624,7 @@ void generate(MethodType methodType) {
             if (useImplMethodHandle) {
                 visitLdcInsn(implMethodCondy);
             }
-            for (int i = 0; i < argNames.length; i++) {
+            for (int i = 0; i < argNames.length - reflectiveCaptureCount(); i++) {
```

Comment:
I stumbled onto this here: https://github.com/openjdk/babylon/pull/53

Reply:
Yes, I know :) Thanks!
We need it here as well.
I'll try to chain to yours.